### PR TITLE
Validationpackage

### PR DESCRIPTION
**Pull request by eduardodir**
## 📥 Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## ☑️ Submission checklist

-   [ ] I have read the [CONTRIBUTING](https://github.com/fremtind/jokul/blob/master/CONTRIBUTING.md) doc
-   [ ] `yarn build` works locally with my changes
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)

## 💬 Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did, what alternatives you considered, etc...

### CODE REVIEW

File: packages/validators-util/package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    "files": [
+        "build"
+    ],
+    "scripts": {
+        "clean": "rimraf node_modules/ .cache/ build/ dist/ public/ **/*.css",
+        "build:types": "tsc -p tsconfig-for-declarations.json",
+        "build:scripts": "rollup --config ../../rollup.config.js",
+        "build": "run-s build:*",
+        "test": "echo \"Error: run tests from root\" && exit 1",
+        "dev:style": "lerna exec --scope=@fremtind/jkl-button yarn build:watch",
```
Comment (lfbergee) [inline packages/validators-util/package.json:31]:
```suggestion
        "dev:style": "echo noop"
```

File: packages/validators-util/src/isEmail/isEmail.test.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { cleanup, render } from "@testing-library/react";
+import { isEmail } from "./isEmail";
+
+describe("isEmail", () => {
+    afterAll(cleanup);
```
Comment (lfbergee) [inline packages/validators-util/src/isEmail/isEmail.test.tsx:6]:
```suggestion
```

File: packages/validators-util/src/isEmail/isEmail.test.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { cleanup, render } from "@testing-library/react";
```
Comment (lfbergee) [inline packages/validators-util/src/isEmail/isEmail.test.tsx:2]:
```suggestion
```

File: packages/validators-util/src/isPhoneNr/isPhoneNr.test.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { cleanup } from "@testing-library/react";
```
Comment (lfbergee) [inline packages/validators-util/src/isPhoneNr/isPhoneNr.test.tsx:1]:
```suggestion
```

File: packages/validators-util/src/isPhoneNr/isPhoneNr.test.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import { isPhoneNr } from "./isPhoneNr";
+
+describe("isPhoneNr", () => {
+    afterAll(cleanup);
```
Comment (lfbergee) [inline packages/validators-util/src/isPhoneNr/isPhoneNr.test.tsx:5]:
```suggestion
```

File: packages/validators-util/documentation/validators.mdx [hunk 6]
```diff
@@ -0,0 +1,24 @@
+---
+title: Validators
+react: validators-util
+---
+
+import IsEmailExample from "./IsEmailExample";
```
Comment (lfbergee) [inline packages/validators-util/documentation/validators.mdx:6]:
```suggestion
import { IsEmailExample } from "./IsEmailExample";
```

File: packages/validators-util/documentation/validators.mdx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import IsPhoneNrExample from "./IsPhoneNrExample";
```
Comment (lfbergee) [inline packages/validators-util/documentation/validators.mdx:7]:
```suggestion
import { IsPhoneNrExample } from "./IsPhoneNrExample";
```

File: packages/validators-util/package.json (hunk 1 above)
Comment (lfbergee) [reply packages/validators-util/package.json:31]: eller bare slette

File: packages/validators-util/src/isPhoneNr/isPhoneNr.ts [hunk 8]
```diff
@@ -0,0 +1,3 @@
+const PHONENR_REGEX = /^\d{8}$/;
+
+export const isPhoneNr = (value: string) => PHONENR_REGEX.test(value);
```
Comment (lfbergee) [inline packages/validators-util/src/isPhoneNr/isPhoneNr.ts:3]: Tror jeg ville kalt den `isPhoneNumber`, jeg syns det er lettere og lese. Koden kan vi minifisere så vi sparer bytene, og intellisense fikser resten, så ingen grunn til å spare på tegnene :dancers:

File: packages/validators-util/src/isPhoneNr/isPhoneNr.test.tsx [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+    it("expect to be CORRECT phone number", () => {
+        expect(isPhoneNr("12345678")).toBe(true);
+    });
+
+    it("expect to be INCORRECT phone number", () => {
```
Comment (lfbergee) [inline packages/validators-util/src/isPhoneNr/isPhoneNr.test.tsx:11]: Antar dette er mest for eksemples skyld, men vi bør ha en haug mer tester for å verifisere validatorene.

Comment (piofinn) [reply packages/validators-util/src/isPhoneNr/isPhoneNr.test.tsx:11]: Tror også jeg ville omforrmulert litt. F.eks. "returns false for incorrect phone numbers", siden det er det som testes at den gjør. Dette er pirk altså 😅

File: packages/validators-util/src/isPhoneNr/isPhoneNr.ts (hunk 8 above)
Comment (eduardodir) [reply packages/validators-util/src/isPhoneNr/isPhoneNr.ts:3]: Jag syns Nr precis som Org, är ganska ganska enkla och förhålla sig till i ett context sammanhang, exempel: OrgNr, den är super vanlig å se i finansiell kode, å alla som jobbar i domänet vet vad det betyder. Jag syns det blir svårare att gå igenom kode, datamodeller när allt må skrivas ut, eg: "NorwegianLimitedOrganisationNumber", hade jag föredragit som NOLimitedOrgNr.  Så länge alla känner till domänet no issues, men de som ikke kan, må förklaras, min synsning är, let it be, let it be :)

File: packages/validators-util/src/isPhoneNr/isPhoneNr.test.tsx (hunk 9 above)
Comment (eduardodir) [reply packages/validators-util/src/isPhoneNr/isPhoneNr.test.tsx:9]: Pirk och inte bara pirk, för att vi må som du säger @lfbergee ha mer tester, validatorer är kritiskt, så mitt förslag, let it be, och optimera, expandera på vägen.

File: packages/validators-util/src/isPhoneNr/isPhoneNr.ts (hunk 8 above)
Comment (piofinn) [reply packages/validators-util/src/isPhoneNr/isPhoneNr.ts:3]: Akkurat "number" er litt spesielt på engelsk; det skal kun forkortes når man snakker om et nummer i en rekke (f.eks. "no. 42"), ikke i setninger. Om man skal forholde seg til den regelen i variabelnavn er jo en annen sak. Dersom vi skal forkorte det foreslår jeg imidlertid å heller bruke `isPhoneNum` så vi ikke blander norsk og engelsk.

File: packages/validators-util/package.json [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    "files": [
+        "build"
+    ],
+    "scripts": {
+        "clean": "rimraf node_modules/ .cache/ build/ dist/ public/ **/*.css",
+        "build:types": "tsc -p tsconfig-for-declarations.json",
+        "build:scripts": "rollup --config ../../rollup.config.js",
+        "build": "run-s build:*",
+        "test": "echo \"Error: run tests from root\" && exit 1",
+        "dev:style": "echo noop"
```
Comment (piofinn) [inline packages/validators-util/package.json:31]: Mangler et komma her
```suggestion
        "dev:style": "echo noop",
```

File: packages/validators-util/documentation/validators.mdx [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+<p>
+    <span class="jkl-lead">Denne pakken innholder validering støttefunksjoner.</span>
```
Comment (lfbergee) [inline packages/validators-util/documentation/validators.mdx:10]: Må det være sånn her? Kan vi ikke sette klassen rett på p-en?

File: packages/validators-util/package.json (hunk 10 above)
Comment (lfbergee) [reply packages/validators-util/package.json:31]: Kanskje like greit å bare slette den, det burde ikke ha noe å si for bygget.

File: packages/validators-util/src/isEmail/isEmail.test.tsx [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
```
Comment (lfbergee) [inline packages/validators-util/src/isEmail/isEmail.test.tsx:1]:
```suggestion
```

File: packages/validators-util/src/isPhoneNr/isPhoneNr.ts (hunk 8 above)
Comment (lfbergee) [reply packages/validators-util/src/isPhoneNr/isPhoneNr.ts:3]: Liker også `num` bedre enn `nr`

File: packages/validators-util/src/isEmail/isEmail.ts [hunk 13]
```diff
@@ -0,0 +1,3 @@
+const EMAIL_REGEX = /^[a-zA-ZæøåÆØÅ0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-ZæøåÆØÅ0-9-]+(?:\.[a-zA-Z0-9-]+)+$/;
```
Comment (lfbergee) [inline packages/validators-util/src/isEmail/isEmail.ts:1]: Burde vi kanskje eksponere regexen også? I dette tilfellet er det jo den som er vikig, ikke testen i seg selv. Så om noen andre ønsker å lage et annet format på validatoren så kan de det.

Comment (frisol97) [reply packages/validators-util/src/isEmail/isEmail.ts:1]: Ikke dumt! Eksponerer regexen også👍🏻

**Comment by lfbergee** [comment]
Er konflikt mot master, ser gjerne at det gjøres en rebase hvor det squases ned til en pen commit.

File: packages/validators-util/documentation/validators.mdx [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+</p>
```
Comment (piofinn) [inline packages/validators-util/documentation/validators.mdx:11]:
```suggestion
<p class="jkl-lead">Denne pakken innholder støttefunksjoner for validering av skjemaer.</span></p>
```

Comment (lfbergee) [reply packages/validators-util/documentation/validators.mdx:11]:
```suggestion
<p class="jkl-lead">
    Denne pakken innholder validering støttefunksjoner.
</p>
```

isåfall.

Comment (piofinn) [reply packages/validators-util/documentation/validators.mdx]: Jeg tenkte ikke over at dette er i `.mdx`-fila. Vi har en egen komponent for ingressen på disse sidene:

```
<Ingress>
    Denne pakken innholder støttefunksjoner for validering av skjemaer.
</Ingress>
```
Trenger ikke importeres noe for å bruke den.

File: packages/validators-util/src/isChassisNumber/isChassisNumber.test.ts [hunk 15]
```diff
@@ -0,0 +1,14 @@
+import { cleanup } from "@testing-library/react";
+import { isChassisNumber } from "./isChassisNumber";
+
+describe("isChassisNumber", () => {
+    afterAll(cleanup);
```
Comment (lfbergee) [inline packages/validators-util/src/isChassisNumber/isChassisNumber.test.ts:5]: Fjern denne.

1. Det brukes ikke @testing-libary her, så hvorfor skulle man trenge å rydde opp etter den?
2. Det er ikke lengre nødvendig å kalle cleanup når man bruker @testing-libary heller, med mindre man gjør rendering manuelt

File: packages/validators-util/src/isChassisNumber/isChassisNumber.test.ts [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+
+    it("expect to be CORRECT chassis number", () => {
```
Comment (lfbergee) [inline packages/validators-util/src/isChassisNumber/isChassisNumber.test.ts:7]: Nå som det faktisk kommer flere ting inn her, så syns jeg vi skal skrive litt mer utfyllende tester. Dette legger grunnlaget for denne pakka, som om vi vil ha decent testdekning her, så bør vi nok starte med litt mer enn dette. Med mindre noen lover på tro og ære at de skal fikse det senere(og faktisk gjør det).

File: packages/validators-util/src/isEmail/isEmail.test.tsx (hunk 2 above)
Comment (lfbergee) [reply packages/validators-util/src/isEmail/isEmail.test.tsx:5]: Denne er ikke resolvet. Please remove line and import.

File: packages/validators-util/src/isPhoneNumber/isPhoneNumber.test.tsx [hunk 17]
```diff
@@ -0,0 +1,14 @@
+import { cleanup } from "@testing-library/react";
+import { isPhoneNumber } from "./isPhoneNumber";
+
+describe("isPhoneNumber", () => {
+    afterAll(cleanup);
```
Comment (lfbergee) [inline packages/validators-util/src/isPhoneNumber/isPhoneNumber.test.tsx:5]: fjern

File: packages/validators-util/src/isRegistrationNumber/isRegistrationNumber.test.ts [hunk 18]
```diff
@@ -0,0 +1,14 @@
+import { cleanup } from "@testing-library/react";
+import { isRegistrationNumber } from "./isRegistrationNumber";
+
+describe("isRegistrationNumber", () => {
+    afterAll(cleanup);
```
Comment (lfbergee) [inline packages/validators-util/src/isRegistrationNumber/isRegistrationNumber.test.ts:5]: fjern

File: packages/validators-util/src/isChassisNumber/isChassisNumber.test.ts (hunk 16 above)
Comment (piofinn) [reply packages/validators-util/src/isChassisNumber/isChassisNumber.test.ts:4]: Enig med dette! Nå er fortsatt de fleste reglene her ganske basic, men er greit å få testet edge cases ordentlig. Synes uansett vi bør formulere navnet på testene bedre. "expect to be correct chassis number" sier ingenting om _hvorfor_ testen skal returnere det den gjør. 

Anbefaler egentlig å skrive alle tingene som skal være oppfylt som en liste, og så bare skrive dem som tester. Det er vel derfor funksjonene heter det de gjør her også ;)

```suggestion
    it("should return true for 9-digit numbers", () => {
```

Comment (lfbergee) [reply packages/validators-util/src/isChassisNumber/isChassisNumber.test.ts:4]: Er ikke bare for testenes skyld heller, tester er ofte den beste måten å finne ut hva som var den eeegentlige intensjonen til koden. Så vi bør ha et sett med positive og negative tester.

File: packages/validators-util/documentation/validators.mdx [hunk 19]
```diff
@@ -0,0 +1,36 @@
+---
+title: Validators
+react: validators-util
+---
+
+import IsEmailExample from "./IsEmailExample";
+import IsPhoneNrExample from "./IsPhoneNrExample";
+import IsPhoneNrExample from "./IsRegistrationNumberExample";
+import IsPhoneNrExample from "./IsChassisNumberExample";
```
Comment (piofinn) [inline packages/validators-util/documentation/validators.mdx:9]: Feil funksjonsnavn i importene

```suggestion
import IsPhoneNumberExample from "./IsPhoneNumberExample";
import IsRegistrationNumberExample from "./IsRegistrationNumberExample";
import IsChassisNumberExample from "./IsChassisNumberExample";
```

Comment (piofinn) [reply packages/validators-util/documentation/validators.mdx:9]: Er også noen feil der de brukes lenger ned, f.eks.:

```tsx
<ComponentExample component={IsRegistrationNumber} /> // mangler "Example"
```

Comment (frisol97) [reply packages/validators-util/documentation/validators.mdx:9]: Haha er det mulig, fikser det

File: portal/gatsby-node.js [hunk 20]
```diff
@@ -1,6 +1,7 @@
 exports.onCreatePage = async ({ page, actions }) => {
     const { createPage, deletePage } = actions;
-    const componentPageMatch = page.path.match(/\/.*react.*\/documentation\/(.*)\//);
+    const componentPageMatch = page.path.match(/\/(.*(react|util).*)\/documentation\/(.*)\//);
```
Comment (piofinn) [inline portal/gatsby-node.js:3]: Tror vi trenger en ny regel under for å håndtere de nye pakkenavnene som blir tatt med. Denne endringen alene vil endre adressen til alle komponentsidene, slik at vi får en haug med døde lenker rundt omkring 😅

Ønsket oppførsel vil være at den genererer path ut fra den delen av pakkenavnet som _ikke_ angir type (f.eks. `-react` eller `-util`), slik at `button-react` blir til `button`, og `validation-util` blir til `validation`.

File: packages/validators-util/documentation/validators.mdx [hunk 21]
```diff
@@ -0,0 +1,36 @@
...
+---
+
+import isValidEpostExample from "./isValidEpostExample";
+import isValidTelefonnummerExample from "./isValidTelefonnummerExample";
+import isValidRegistreringsnummerExample from "./isValidRegistreringsnummerExample";
+import isValidChassisnummerExample from "./isValidChassisnummerExample";
+
+<Ingress>Denne pakken innholder validering støttefunksjoner.</Ingress>
+
+## isValidEpost
```
Comment (lfbergee) [inline packages/validators-util/documentation/validators.mdx:13]: Er epost et norsk domene ord? :stuck_out_tongue:

File: packages/validators-util/package.json (hunk 1 above)
Comment (lfbergee) [reply packages/validators-util/package.json:31]: Tja, denne ligger fortsatt her. Gjør ikke så veldig mye at den er der, bare litt rot.

File: packages/validators-util/src/isValidRegistreringsnummer/isValidRegistreringsnummer.ts [hunk 22]
```diff
@@ -0,0 +1,3 @@
+export const REGNR_REGEX = /^[a-zA-Z]{2}\d{5}/;
+
+export const isValidRegistreringsnummer = (value: string) => REGNR_REGEX.test(value);
```
Comment (lfbergee) [inline packages/validators-util/src/isValidRegistreringsnummer/isValidRegistreringsnummer.ts:3]: Nå jobber ikke jeg med bil, men er ikke denne for lite spesifik? Eller er tanket at Registeringsnummer er default for vanlig bil. Også får man mer spesifike for feks RegisteringsnummerMC, veteran osv?

File: portal/gatsby-node.js (hunk 20 above)
Comment (lfbergee) [reply portal/gatsby-node.js:3]: Blir konflikt med colors branchen, løst litt aneledes der, så kan evt fikse det når den merges om denne går først.

File: packages/validators-util/src/isValidRegistreringsnummer/isValidRegistreringsnummer.ts (hunk 22 above)
Comment (frisol97) [reply packages/validators-util/src/isValidRegistreringsnummer/isValidRegistreringsnummer.ts:3]: Dette er bare for bil ja, burde ha spesifisert det. Er det best å ha alt i en regex eller lage spesifikke regler for hvert kjøretøy?

**Comment by piofinn** [comment]
Oops! Klarte å rebase med gammel versjon av `master` 😭 Skal fikse!
